### ISSUE TITLE
FirebasePerformance crash for separate process java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process

### ISSUE THREAD

**Issue by alexklints** [firebase/firebase-android-sdk#6039]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: _____
  * Firebase Component: `com.google.firebase:firebase-perf`
  * Component version: `com.google.firebase:firebase-bom:33.1.0` (Reproduced since `com.google.firebase:firebase-bom:31.5.0`)

### [REQUIRED] Step 3: Describe the problem
Calling `URL.openConnection()` from **separate process** will lead crash with Firebase Performance. 
Stacktrace
```
FATAL EXCEPTION: DefaultDispatcher-worker-2
Process: com.alexklints.firebaseperformanceissue:Second, PID: 7698
java.lang.ExceptionInInitializerError
	at com.google.firebase.perf.config.ConfigResolver.<init>(ConfigResolver.java:78)
	at com.google.firebase.perf.config.ConfigResolver.getInstance(ConfigResolver.java:86)
	at com.google.firebase.perf.application.AppStateMonitor.<init>(AppStateMonitor.java:98)
	at com.google.firebase.perf.application.AppStateMonitor.getInstance(AppStateMonitor.java:87)
	at com.google.firebase.perf.metrics.NetworkRequestMetricBuilder.<init>(NetworkRequestMetricBuilder.java:92)
	at com.google.firebase.perf.metrics.NetworkRequestMetricBuilder.builder(NetworkRequestMetricBuilder.java:84)
	at com.google.firebase.perf.network.FirebasePerfUrlConnection.instrument(FirebasePerfUrlConnection.java:189)
Caused by: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process com.alexklints.firebaseperformanceissue:Second. Make sure to call FirebaseApp.initializeApp(Context) first.
	at com.google.firebase.FirebaseApp.getInstance(FirebaseApp.java:179)
	at com.google.firebase.perf.config.RemoteConfigManager.getInitialStartupMillis(RemoteConfigManager.java:92)
	at com.google.firebase.perf.config.RemoteConfigManager.<init>(RemoteConfigManager.java:86)
	at com.google.firebase.perf.config.RemoteConfigManager.<clinit>(RemoteConfigManager.java:53)
	... 16 moreFATAL EXCEPTION: DefaultDispatcher-worker-2
```

#### Steps to reproduce:
1. [Add Firebase Perf to project](https://firebase.google.com/docs/perf-mon/get-started-android?hl=en).
2. Make network request with UrlConnection from activity.
3. All works.
4. Than move activity to separate process. Activity just for simple example. It can be a service in a separate process.
5. Crash occurs. 
6. Expected behavior: Requests will be executed correctly in a separate process without activating FirebaseApp. 

It works correct up to `com.google.firebase:firebase-bom:31.0.0`. 

#### Relevant Code:

[Example with network request from separate process](https://github.com/user-attachments/files/15855790/FirebasePerfomanceIssue.zip)

**Comment by lehcar09** [comment]
Hi @alexklints, thank you for for reaching out and raising the issue. Thank you for sharing an MCVE. I was able to reproduce the issue from version 31.2.0 up to the latest version. I'll notify our engineers about and see what we can do here. Thanks!

**Comment by mrober** [comment]
Firebase only initializes automatically on the default process of an app. If you have a multi-process app, you need to initialize manually in each process or not use Firebase in the other processes.

To initialize manually, you could do something like this:
```kotlin
class MyApp : Application() {
  override fun onCreate() {
    super.onCreate()

    // Initialize Firebase in other processes.
    if (getProcessName() != applicationInfo.processName) {
      FirebaseApp.initializeApp(applicationContext)
    }
  }
}
```

**Comment by alexklints** [comment]
Hi. 

I understand, but the thing is, the behavior used to be different. Firebase Performance did not attempt to intercept the network if there was no initialization in the process. There are such examples of an issue where developers tried to implement this behavior: 
- https://github.com/firebase/firebase-android-sdk/pull/5616/files
- https://github.com/firebase/firebase-android-sdk/issues/5584
In addition, many developers, unfortunately, are very little familiar with the concept of a process. They add a library that adds a process to the application. And everything starts to fall for them. 
In addition, Firebase is not well adapted to multi-process applications. Crashes and ANR occur periodically when trying to initialize Firebase and its individual components in more than one process. 
I believe that if you fix this issue, it will become easier for many developers.

**Comment by mrober** [comment]
I took a look, and this was fixed in Firebase Perf 21.0.1, can you try the latest version @alexklints?

**Comment by mrober** [comment]
Please open a new issue if this was not resolved in the latest version for you.
